### PR TITLE
Add tips for review club hosts

### PR DESCRIPTION
**Pull request by jnewbery**
We've had quite a few guest hosts for the PR review club. I wanted to collect together some of what we've learned so any future hosts can benefit from our experience.

This PR adds a new CONTRIBUTING.md file with some general tips for hosting.

### CODE REVIEW

**Comment by jnewbery** [comment]
If @ariard @harding @jachiang @jkczyz @jonatack @MarcoFalke @mzumsande or @pinheadmz have any additional tips, I'd love to hear them!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    for the host to ask a specific question - just jump in at any point.
+- keep an eye on the clock, and try to wrap up the meeting at the end of the
+  hour. Even if you're happy to continue beyond the hour, some of the attendees
+  might have hard stops.
+- be encouraging! For many people, asking questions can be intimidating, even
+  in the pseudonymous comfort of IRC! People are there to learn. Try to create
+  an environment where they feel safe to ask any questions and where they can
+  try to answer.
+- have fun, and pat yourself on the back for making Bitcoin protocol development
+  stronger and more decentralized 🚀
```
Comment (ariard) [inline CONTRIBUTING.md]: Following the type of PR (P2P, RPC, threading refactor) a review is not going to be the same and I think people are really interested by your methodology so could be cool to talk about it like using `--enable-debug` for running tests for a PR with thread changes or interesting options like `-noconnect`  or understanding first the conceptual problems solved for a P2P one

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
```
Comment (ariard) [inline CONTRIBUTING.md]: If the PR is  one part of a serie, it's really interesting to do a bit of archeology and provide pointers to previous ones, especially if they're spanned on multiple years

**Comment by jonatack** [comment]
Excellent idea! I'll circle back here in a couple of days, but I'd like to emphasize how useful the study notes can be. When I revisit these review club PRs months later, some of the study notes really stand out in value -- an example of this today when I went back over https://bitcoincore.reviews/15681.html by @harding -- a high bar for writing, to be sure, but evergreen. @amitiuttarwar's notes for tomorrow look good, too.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+- don't be upset if people haven't had a chance to review the PR yet. This is a
+  voluntary meeting and most people have a lot of other commitments.
+- when hosting, it sometimes feels like there are minutes of silence and no-one
+  is out there. Be patient and don't worry! It takes people a bit of time to
+  formulate their thoughts and type them out.
+- it's nice to remind people of some of our meeting conventions at the start:
+  - you don't have to ask to ask a question (eg "I have a question about x but
+    I don't know if it's on-topic?"). Just go ahead and ask. If it's off-topic
+    we'll tell you.
+  - the host is there to help moderate, not to lead. You don't need to wait
```
Comment (jonatack) [inline CONTRIBUTING.md]: Interesting point about moderating, not leading.

Perhaps mention to the PR author that their PR will be a topic of the next session and invite them to join in... they just might! Their perspective can be interesting for everyone and helpful.

Also, investing real time preparing your meeting, if you are able to, can pay off in unexpected ways even months later.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Hosting a Bitcoin Core review club meeting
+
+## General Tips
+
+Some tips for hosting a review club meeting:
+
+- try to post notes and questions on the Friday the week before the meeting.
+  That gives attendees time to review the PR and prepare for the meeting.
+- prepare a range of questions from basic to advanced. The review club is a way
+  to help new developers develop. There should be something for everone.
```
Comment (amitiuttarwar) [inline CONTRIBUTING.md]: nit: everyone

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+- come with the mindset that the attendees are people who are as smart as you,
+  but don't have the context that you do in Bitcoin protocol development.
+- very open questions to the group (eg "Does everyone know how tx gossip works
+  in Bitcoin?") sometimes fall flat because all the attendees are waiting for
+  someone else to answer. Instead try to make the questions focussed on the
```
Comment (amitiuttarwar) [inline CONTRIBUTING.md]: nit: focused

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  PR or change set that is being reviewed (eg "how does commit X change the
+  way bitcoind gossips transactions?")
```
Comment (amitiuttarwar) [inline CONTRIBUTING.md]: in addition to specificity, phrasing the question positively (eg. "can anybody describe how...") encourages people to speak up vs negatively (eg. "does anybody not understand...")

File: CONTRIBUTING.md (hunk 4 above)
Comment (jkczyz) [inline CONTRIBUTING.md]: To reiterate what @jonatack said, putting in the effort to prep for the meeting will really pay off. I found it useful to have answers to all the questions written down in advanced, especially when there are multiple correct answers and code references to give. A participant may get the answer sorta right, so having text readily available to make use of either directly or as an aid in responding will make the session run smoothly.

File: CONTRIBUTING.md (hunk 5 above)
Comment (jonatack) [reply CONTRIBUTING.md]: In British/Australian/New Zealander English, it's not an error AFAIK. https://www.collinsdictionary.com/dictionary/english/focussed

**Comment by jonatack** [comment]
More on the study notes: I think it can be good to provide DIY hints, e.g. suggestions for things to git grep, tests to modify, compile flags to pass, asserts/printfs/logging to add, gdb/pbd/strace/valgrind examples, bdb commands like dbdump to inspect wallet files, etc.

**Comment by pinheadmz** [comment]
Great list! Only nit: to my eye, would prefer each bullet point be capitalized (they are sentences). Also the title of the club is Bitcoin Core Review Club, should be Title Case [here](https://github.com/bitcoin-core-review-club/bitcoin-core-review-club.github.io/pull/71/files#diff-6a3371457528722a734f3c51d9238c13R1).

File: CONTRIBUTING.md (hunk 5 above)
Comment (amitiuttarwar) [reply CONTRIBUTING.md]: TIL

**Comment by jnewbery** [comment]
Thanks for your input everyone. I've pushed a second commit which I think incorporates all of your suggestions.

**Comment by pinheadmz** [comment]
ACK 4507ad14e966a4ee266c8425e8fdea5039f3e5f3

With one additional, optional thought: Would you want to add a link to the website as well?
"Interested in hosting a Review Club meeting? Click here..."

**Comment by jnewbery** [comment]
> Would you want to add a link to the website as well?
"Interested in hosting a Review Club meeting? Click here..."

Great idea. Done!

**Comment by jnewbery** [comment]
This is now merged, but if anyone has any further suggestions for the host guide, feel free to open a PR or issue.

Thanks for the review and suggestions 🙏

**Pull request merged**